libs/cardano-ledger-binary/test/Test/Cardano/Ledger/Binary/Success.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Cardano.Ledger.Binary.Success (spec) where
+
+import Cardano.Ledger.Binary
+import Data.Either (isRight)
+import Data.List (nub)
+import Data.Set (Set)
+import Test.Cardano.Ledger.Binary (decode)
+import Test.Hspec
+import Test.Hspec.QuickCheck
+import Test.QuickCheck
+
+-- | Generate a list with unique elements
+genUniqueList :: Gen [Int]
+genUniqueList = nub <$> arbitrary
+
+-- | Generate a CBOR encoded list with no duplicates, with and with the set tag
+genUniqueListEncoding :: Gen Encoding
+genUniqueListEncoding = do
+  xs <- genUniqueList
+  let definite = encodeListLen (fromIntegral $ Prelude.length xs) <> foldMap encCBOR xs
+      indefinite = encodeListLenIndef <> foldMap encCBOR xs <> encodeBreak
+  oneof
+    [ pure $ definite
+    , pure $ encodeTag 258 <> definite
+    , pure $ indefinite
+    , pure $ encodeTag 258 <> indefinite
+    ]
+
+-- | Starting in version 9, check set decoding with and without tag 258
+prop_shouldFailSetWithDupKeys :: Property
+prop_shouldFailSetWithDupKeys =
+  forAllBlind genUniqueListEncoding $
+    ( \setEncoder ->
+        (property . isRight) (decode (natVersion @9) setEncoder :: Either DecoderError (Set Int))
+    )
```

Comment:
`isRight` isn't that great for debugging when failure occurs :smile: 
```suggestion
     \setEncoder ->
       let trip = Trip id (decCBOR @(Set Int))
       in property $ embedRoundTrip trip (natVersion @9) maxBound setEncoder
```

Reply:
> `isRight` isn't that great for debugging when failure occurs smile

you missed the opportunity for a good pun:

> `isRight` isn't that great for debugging when failure occurs, amiright?

Reply:
How about this one? :smile: 

`isRight` isn't right for debugging